### ISSUE TITLE
Error in fedora 39. 'Photoshop/path/' is a 32-bit installation, it cannot support 64-bit applications.

### ISSUE THREAD

**Issue by nasawyer7** [LinSoftWin/Photoshop-CC2022-Linux#126]
**PLEASE READ THE REQUIREMENT BEFORE FILLING A BUG. ALSO PLEASE CHECK IF A SIMILLAR ISSUE ISN'T OPEN**

**Photoshop version**
eg : CC2022

**Describe the bug**

**To Reproduce**
I'm on fedora 39, and in order to get wine 7.22, i had to compile from source. I downloaded wine from the main website, and used ./configure CFLAGS="-I/usr/include/freetype2", since freetype2 wasn't working. Anyways, made it normally after that, and wine --version functions just fine, and it returnes 7.22.

I tried to install 2022 with command line, and I got 
nathan@fedora:~/Downloads/wine-7.22$ /home/nathan/Music/Adobe-Photoshop/drive_c/launcher.sh %F
free(): invalid pointer
0044:err:seh:NtRaiseException Exception frame is not in stack limits => unable to dispatch exception.
0084:fixme:hid:handle_IRP_MN_QUERY_ID Unhandled type 00000005
0084:fixme:hid:handle_IRP_MN_QUERY_ID Unhandled type 00000005
0084:fixme:hid:handle_IRP_MN_QUERY_ID Unhandled type 00000005
0084:fixme:hid:handle_IRP_MN_QUERY_ID Unhandled type 00000005
wineserver: using server-side synchronization.
wine: '/home/nathan/Music//Adobe-Photoshop' is a 32-bit installation, it cannot support 64-bit applications.

Any ideas?
**Expected behavior**

**Screenshots**

**System informations**
- Distro :Fedora 39
- GPU : (AMD)
- WINE version : 7.22 

**Additional context**

**Is some information are missing it might take longer to find the issue so please complete carefully**

**Comment by MiMillieuh** [comment]
Please just use the latest wine versions and it should be fine. compiling wine is a headache. if you use wine-devel it should be ok.

If you can't install wine for some reasons, you can use one of my builds. they works but they aren't perfect... : https://github.com/MiMillieuh/silverwine (Please note that these builds doesn't work with illustrator yet for dependencies reasons.)

**Comment by nasawyer7** [comment]
I previously had the fedora 9.0 wine installed, and kept on running into [this](https://github.com/LinSoftWin/Photoshop-CC2022-Linux/issues/94#issuecomment-1426776219) issue. Will your wine build fix the windows ten bug?

**Comment by MiMillieuh** [comment]
It should be fixed in 9.0 builds... did you removed the prefix before reinstalling ?

**Comment by nasawyer7** [comment]
yes, just removed ~/.wine and the folder i previously attempted to install too while installing using the default wine 9.0 from fedora, and I still run into that windows error.

**Comment by MiMillieuh** [comment]
Well 2021 is less buggy but if you encounter a win 10 issue I guess it's not really related.

can you share what hardware you're using in details ?

**Comment by nasawyer7** [comment]
Yeah, I'm on ryzen 9 6900hs, amd 6700s (gpu), 40 gb ram, fedora 39 kde. Anything else I should add? I never touched wine before this project.

**Comment by MiMillieuh** [comment]
Not really, I just wanted to be certain that you're on a 64bit system.

Did you installed wine from the winehq repo ?

**Comment by nasawyer7** [comment]
no, i just did sudo dnf install wine, and then sudo dnf install wine-devel. When I run wine --version, i get wine-9.0 (Staging)

**Comment by MiMillieuh** [comment]
Can you install only the devel version and not the staging version as it's known to cause issues with PS
https://wiki.winehq.org/Fedora

**Comment by nasawyer7** [comment]
Yeah for sure, just installed the devel version from the winehq repo. I still ran into the error, even after i removed ~/.wine before installing. Am i doing something wrong? or leaving around some file I haven't deleted yet from an old wine?

**Comment by MiMillieuh** [comment]
I don't really know... I can't access your machine 😅 

May you can tell me how you're install it ? and how you're installing PS ?

**Comment by nasawyer7** [comment]
I just ran git clone to download the files, and then i tried installing everything in the scripts folder. I did not run anything as sudo, just sh scriptname filepath. Both the 2021 and 2022 scripts returned the same issue. Each time I would install to a different folder.

For wine, I just have installed winehq-devel from the wine repository. Wine --version returns wine-9.1.

I think the issue is that it's not using the correct wineprefix. it keeps on defaulting to the wineprefix at .wine (I know this since I can remove it, and it gets repopulated when I run it again.). 
So, I tried to the export WINEPREFIX=/path/too/install/, but now i get the error: 
wine client error:0: version mismatch 786/755.
Your wine binary was not upgraded correctly,
or you have an older one somewhere in your PATH.
Or maybe the wrong wineserver is still running?

**Comment by MiMillieuh** [comment]
Can you try to run it in a clear environment ? like a VM or a distrobox ? 

These errors looks like if a wine version has been downgraded and you still try to use your prefix... that's not the case but that's a strange behaviour.

**Comment by nasawyer7** [comment]
Alright, just ran it in kvm with a fedora 39 iso. Updated it fully, and then ran it. I installed wine using wine-devel, and when I run wine --version, I got wine-9.1(staging). I will retry with the official wine repository, but i doubt that the outcome will be any different.

Ran into the windows ten error again on system wine.

**Comment by RyzenDew** [comment]
it installed fine on fedora 39 i did it yesterday

**Comment by nasawyer7** [comment]
> it installed fine on fedora 39 i did it yesterday

Which version of wine did you use?

**Comment by rickywid** [comment]
> Which version of wine did you use? I just tried it on the vm and it did not work. Would it matter that I'm in x11? Or that i'm in kde?

Not sure if you got it to work, if not,  I suggest using this [repo](https://github.com/Gictorbit/photoshopCClinux/) instead. I was struggling to get photoshopcc2021 repo to install as well after countless hours wasted but then I tried the other repo and it worked on my first attempt.

Unless you are dead set on using PhotoshopCC2021, you should try the other repo that I linked.

**Comment by nasawyer7** [comment]
Yeah, now im having trouble running into an error that says 
curl: (35) OpenSSL/3.1.1: error:0A000438:SSL routines::tlsv1 alert internal error
tar: This does not look like a tar archive

I tried the silverwine, same issue. The reason why I tried this repo is because the other repo had other issues, and it did not work on first attempt at all

**Comment by gmankab** [comment]
you can try flatpak

- [photoshop2021](https://github.com/gmanka-flatpaks/com.adobe.photoshop2021)
- [photoshop2022](https://github.com/gmanka-flatpaks/com.adobe.photoshop2022)
